zapisy/apps/grade/poll/views.py

```diff
@@ -182,6 +185,51 @@ def __get_counter_for_categories(polls):
 
         return number_of_submissions_for_category
 
+    @staticmethod
+    def __are_read(polls, user):
+        is_read_category = defaultdict(True.__bool__)
+        is_read_poll = dict()
+
+        last_views: Dict[Poll, datetime.datetime] = dict(
+                PollView.objects.filter(user=user, poll__in=polls).values_list("poll", "time")
+            )
+
+        last_modifieds = Submission.objects.filter(poll__in=polls)
+        for poll in polls:
+            if poll.id in last_views:
+                try:
+                    last_modified = last_modifieds.filter(poll=poll).latest('modified')
+                    is_read_poll[poll] = (last_views[poll.id] > last_modified.modified)
+                except Submission.DoesNotExist:
+                    is_read_poll[poll] = True
+            else:
+                val = all([not sub.submitted for sub in last_modifieds.filter(poll=poll)])
+                is_read_poll[poll] = val
+            is_read_category[poll.category] &= is_read_poll[poll]
+
+        return [is_read_category, is_read_poll]
+
+    @staticmethod
+    def __are_viewed_answers(submissions, user):
+        if submissions:
+            try:
+                last = PollView.objects.get(user=user, poll=submissions[0].poll)
+            except PollView.DoesNotExist:
+                last = None
+            viewed = dict()
+            for submission in submissions:
+                if 'schema' in submission.answers:
```

Comment:
```suggestion
                if 'schema' not in submission.answers:
                    continue
```
a potem reszta leci bez dodatkowego zagnieżdżenia, które tutaj dość szybko nam przyrastają.

Reply:
W tej "reszcie" trzeba zmniejszyć zagnieżdżenie, w tej chwili jest ona za `continue` i jest nieosiągalna.